iotc-generic-c-sdk/src/iotconnect.c

```diff
@@ -227,16 +228,97 @@ bool iotconnect_sdk_is_connected(void) {
     return iotc_device_client_is_connected();
 }
 
-IotConnectClientConfig *iotconnect_sdk_init_and_get_config(void) {
+//
+// Reset the config
+//
+// Deallocate any dynamically allocated data
+// Disconnect MQTT, clear any previous discovery and sync responses -- as may reconfigure to a new device.
+//
+void iotconnect_sdk_reset_config(void) {
+    if(iotconnect_sdk_is_connected() == true) {
+        iotconnect_sdk_disconnect();
+    }
+
+    tpm_registration_id = NULL;
+
+    iotcl_discovery_free_sync_response(sync_response);
+    sync_response = NULL;
+
+    iotcl_discovery_free_discovery_response(discovery_response);
+    discovery_response = NULL;
+
+    if(config.dynamic_memory) {
+        free(config.cpid);
+        free(config.duid);
+        free(config.env);
+        free((void *) config.auth_info.data.symmetric_key);
```

Comment:
Very minor note: No need to cast to (void*). Any compiler should accept a cast of char* to void * as long as the pointer is not const. Casting the symmetric key to char* would be the correct cast in my opinion.